### ISSUE TITLE
Question: Is there a way to skip nuitka compilation?

### ISSUE THREAD

**Issue by scopatz** [Nuitka/Nuitka#650]
For example, say I have some code that I want to use with CPython but don't want nuitka to see.  How do I avoid code blocks?  Is there a decorator or a special comment? For example, something like:

```python
@nuitka_skip
def my_cpython_func():
    ...

# if not nuitka
cpython_variable = 42
# endif
```

**Comment by kayhayen** [comment]
This is a valid request, and Nuitka doesn't support it yet. For a whole module, you can do a `--nofollow-module` and then it won't include it. You have to make it available in the `PYTHONPATH` which is the normal one, or through the trick that is intended for test packages, via special environment variables.

Per function decorations that would e.g. preserve bytecode for them, that's not there.

You can check the `__compiled__` attribute though, and branch dependent on that, that way, you can make the variable different. Of course that is horrible, and you better ought not ask our support, if the nuitka compiled program behaves differently, as we won't debug these differences and their effects.

**Comment by scopatz** [comment]
Yeah, I am mostly looking for the ability to skip some globals-modifying state code that I have for CPython performance improvement. Namely, I would like to make [lazyasd](https://github.com/xonsh/lazyasd) skip all of its actual behavior under nuitka

**Comment by anki-code** [comment]
I've found a way how to build static linked binary portable python app using Nuitka. Docker example you can find in https://github.com/xxh/xxh-portable . The main feature is avoiding using glibc and use static linked musle libc instead. It works on many linux.

@scopatz if you can suggest workaround for the lazyasd exception I'll be happy to try build xonsh using this technique.

**Comment by scopatz** [comment]
@anki-code, that's awesome!  Yeah.  Let's try to find a way to work around lazyasd.  I think would could probably try to find a way to strip it out of the code base entirely.  We probably also don't need to use amalgamate either, but it may or may not make the result of Nuitka faster.  We should figure that out.

**Comment by anki-code** [comment]
Hello @kayhayen! Thank you for nuitka it runs fast!
Just a quick question: half of the year passed and maybe appeared some additional thoughts or plans to make the enhancement described in first message of this thread?

**Comment by kayhayen** [comment]
There is a bit of an improvement in sight here, Nuitka now can do it based on decorator names. The code is still experimental, and has only been applied for the `pyQtslot` decorator so far, still rough, but should pave a way to this feature request to eventually become true. Setting this to a 1.8 milestone, but not sure I can make it in time. Most difficult part is to decide how to name it and where to put it. Will need a new package on PyPI to provide tools like this one. I am tempted to use `nuitka_hints` or something like that as a package name.

**Comment by kayhayen** [comment]
I am shifting this to 2.2 release, as it really needs a different solution than what we did for PyQt5/6 there really, as that's causing issues long term, the solution needs to happen at code generation time, and not at tree building I now believe, such that the code does still get analysed properly and can have effects, like following imports, metadata versions, etc.

**Comment by kayhayen** [comment]
I have implemented it, plugins now can provide a list of names, and "nuitka_skip" will be added hard coded. Need to add documentation for this somewhere still, but I don't track that here.

**Comment by anki-code** [comment]
@kayhayen thanks!!! If you can drop an example here I can try it in xonsh shell.

**Comment by kayhayen** [comment]
Thanks for your report. This is fixed on the factory branch, which is a development version that is under rapid development. You can try it out by going here: https://nuitka.net/doc/factory.html 
	
Feedback on whether this is working is very welcome. Please do not share plans to do it; only confirm or deny that it is working.

**Comment by kayhayen** [comment]
```
def nuitka_ignore(func): return func

@nuitka_ignore
def my_cpython_func():
    ...

```

I haven't tested it, but importing it from another module and doing `some_module.nuitka_ignore` is supposed to work, and using it as a callable (like pyQtSlot), too. Let me know if that works for you. A pre-processor for the source code, if not what I want to do, people can get for the `__compiled__` value if they really need that, but mind you, I am pretty sure it ought to never be necessary to do that.

**Comment by kayhayen** [comment]
This is part of the stable release 4.0 that I just made.
